src/lib/openapi/spec/create-api-token-schema.ts

```diff
@@ -1,68 +1,83 @@
 import { FromSchema } from 'json-schema-to-ts';
-import { ApiTokenType } from '../../types/models/api-token';
 
 // TODO: (openapi) this schema isn't entirely correct: `project` and `projects`
 // are mutually exclusive.
 //
 // That is, when creating a token, you can provide either `project` _or_
 // `projects`, but *not* both.
 //
-// We should be able to annotate this using `oneOf` and `allOf`, but making
-// `oneOf` only valid for _either_ `project` _or_ `projects` is tricky.
-//
-// I've opened an issue to get some help (thought it was a bug initially).
-// There's more info available at:
-//
-// https://github.com/ajv-validator/ajv/issues/2096
-//
-// This also applies to apiTokenSchema and potentially other related schemas.
-
+// Because we allow additional properties, we cannot express the mutual
+// exclusiveness in the schema (with OpenAPI 3.0). As such, it's mentioned in
+// the description for now.
 export const createApiTokenSchema = {
     $id: '#/components/schemas/createApiTokenSchema',
     type: 'object',
-    required: ['type'],
+    description:
+        'The data required to create an [Unleash API token](https://docs.getunleash.io/reference/api-tokens-and-client-keys).',
     properties: {
-        secret: {
-            type: 'string',
-        },
-        type: {
-            type: 'string',
-            description: `One of ${Object.values(ApiTokenType).join(', ')}`,
-        },
-        environment: {
-            type: 'string',
-        },
-        project: {
+        tokenName: {
             type: 'string',
-        },
-        projects: {
-            type: 'array',
-            items: {
-                type: 'string',
-            },
+            description: 'The name of the token.',
+            example: 'client-token-64522',
         },
         expiresAt: {
             type: 'string',
             format: 'date-time',
-            nullable: true,
+            description: 'The time when this token should expire.',
+            example: '2023-07-04T11:26:24+02:00',
+        },
+        username: {
+            deprecated: true,
+            type: 'string',
+            description:
+                'The name of the token. This property is deprecated. Use `tokenName` instead.',
+            example: 'client-token-64523',
         },
     },
-    anyOf: [
+    oneOf: [
         {
+            required: ['type'],
             properties: {
-                username: {
+                type: {
                     type: 'string',
+                    pattern: '[Aa][Dd][Mm][Ii][Nn]',
```

Comment:
Would /admin/i work?

Reply:
In a typical regex syntax it should work but I'm not sure about the openapi engine